File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/WireTransfer/WireTransferFeatures.cs
namespace Psi.Data.Models.ClientConfigurationModels.WireTransfer
{
    public class WireTransferFeatures : SettingsBaseHelper
    {
        public WireTransferFeatures(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        /// <summary>
        /// Enables wire transfers with secure messages 2
        /// </summary>
        [SettingKey("WireTransfer.Enabled")]
        public bool Enabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        /// <summary>
        /// min version
        /// </summary>
        [SettingKey("WireTransfer.MinVersion")]
        public double MinVersion
        {
            get { return GetDoubleValue(); }
            set { SetValue(value); }
        }

        /// <summary>
        /// Wire Transfer Department 'to' Email Address
        /// </summary>
        [SettingKey("X.App.HomeBanking.WireTransferDeptEmail")]
        public string WireTransferDepartmentToEmail
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        /// <summary>
        /// Wire Transfer Department 'From' Email Address
        /// </summary>
        [SettingKey("X.App.HomeBanking.gsEmailFrom")]
        public string WireTransferDepartmentFromEmail
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }
    }
}

File: infrastructure/models/ClientConfigurationModels/WireTransfer/WireTransferFeatures.ts
import { Setting, ISettingsGroup, ISettingsMetadata } from '@models/base/types';
export interface WireTransferFeaturesConfig {
    Enabled: boolean;
    MinVersion: number;
    WireTransferDepartmentToEmail: string;
    WireTransferDepartmentFromEmail: string;
}

export class WireTransferFeatures implements ISettingsGroup {
    static readonly metadata: ISettingsMetadata = {
        groupName: 'WireTransferFeatures'
    };


            private _enabled: boolean;
            get enabled(): boolean {
                return this._enabled;
            }
            set enabled(value: boolean) {
                this._enabled = value;
            }

            private _minVersion: number;
            get minVersion(): number {
                return this._minVersion;
            }
            set minVersion(value: number) {
                this._minVersion = value;
            }

            private _wireTransferDepartmentToEmail: string;
            get wireTransferDepartmentToEmail(): string {
                return this._wireTransferDepartmentToEmail;
            }
            set wireTransferDepartmentToEmail(value: string) {
                this._wireTransferDepartmentToEmail = value;
            }

            private _wireTransferDepartmentFromEmail: string;
            get wireTransferDepartmentFromEmail(): string {
                return this._wireTransferDepartmentFromEmail;
            }
            set wireTransferDepartmentFromEmail(value: string) {
                this._wireTransferDepartmentFromEmail = value;
            }


    constructor() {}


        toSettings(): Setting[] {
            return [
                { key: "WireTransferFeatures.Enabled", value: this._enabled, dataType: 'boolean', label: "Enabled" },
                { key: "WireTransferFeatures.MinVersion", value: this._minVersion, dataType: 'number', label: "Min Version" },
                { key: "WireTransferFeatures.WireTransferDepartmentToEmail", value: this._wireTransferDepartmentToEmail, dataType: 'string', label: "Wire Transfer Department To Email" },
                { key: "WireTransferFeatures.WireTransferDepartmentFromEmail", value: this._wireTransferDepartmentFromEmail, dataType: 'string', label: "Wire Transfer Department From Email" },
            ];
        }

}